x-pack/plugins/cases/public/components/files/files_table.tsx

```diff
@@ -0,0 +1,189 @@
+  const { client: filesClient } = useFilesContext();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [selectedFile, setSelectedFile] = useState<FileJSON>();
+
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = (file: FileJSON) => {
+    setSelectedFile(file);
+    setIsModalVisible(true);
+  };
+
+  const columns: Array<EuiBasicTableColumn<FileJSON>> = [
+    {
+      name: i18n.NAME,
+      'data-test-subj': 'files-table-filename',
+      render: (attachment: FileJSON) => (
+        <EuiLink onClick={() => showModal(attachment)}>{attachment.name}</EuiLink>
+      ),
+      width: '60%',
+    },
+    {
+      name: i18n.TYPE,
+      field: 'mimeType',
+      'data-test-subj': 'files-table-filetype',
+    },
+    {
+      name: i18n.DATE_ADDED,
+      field: 'created',
+      'data-test-subj': 'files-table-date-added',
+      dataType: 'date',
+    },
+    {
+      name: i18n.ACTIONS,
+      width: '120px',
+      actions: [
+        {
+          name: 'Download',
+          isPrimary: true,
+          description: i18n.DOWNLOAD_FILE,
+          render: (attachment: FileJSON) => {
+            return (
+              <EuiButtonIcon
+                iconType={'download'}
+                aria-label={'download'}
+                href={filesClient.getDownloadHref({
+                  fileKind: CASES_FILE_KINDS[APP_ID].id,
+                  id: attachment.id,
+                })}
+              />
+            );
+          },
+          'data-test-subj': 'files-table-action-download',
+        },
+        {
+          name: 'Delete',
+          isPrimary: true,
+          description: i18n.DELETE_FILE,
+          color: 'danger',
+          icon: 'trash',
+          type: 'icon',
+          onClick: () => {},
+          'data-test-subj': 'files-table-action-delete',
+        },
+      ],
+    },
+  ];
+
+  const resultsCount = useMemo(
+    () => (
+      <>
+        <strong>
+          {pagination.pageSize * pagination.pageIndex + 1}
+          {'-'}
+          {pagination.pageSize * pagination.pageIndex + pagination.pageSize}
+        </strong>{' '}
+        {'of'} <strong>{pagination.totalItemCount}</strong>
+      </>
+    ),
+    [pagination.pageIndex, pagination.pageSize, pagination.totalItemCount]
+  );
+
+  return isLoading ? (
+    <EuiLoadingContent data-test-subj="files-table-loading" lines={10} />
+  ) : (
+    <>
+      {pagination.totalItemCount > 0 && (
+        <>
+          <EuiSpacer size="xl" />
+          <EuiText size="xs" data-test-subj="files-table-results-count">
+            {i18n.RESULTS_COUNT} {resultsCount}
+          </EuiText>
+        </>
+      )}
+      <EuiSpacer size="s" />
+      <EuiBasicTable
+        tableCaption="Files Table"
+        items={items}
+        columns={columns}
+        pagination={pagination}
+        onChange={onChange}
+        data-test-subj="attachments-table"
+        noItemsMessage={<EmptyFilesTable />}
+      />
+      {isModalVisible && (
```

Comment:
Should we add in a check to ensure that `selectedFile !== undefined` or `selectedFile != null`?